TPP.Core/Commands/Definitions/PollCommands.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using TPP.ArgsParsing.Types;
+using TPP.Persistence.Models;
+using TPP.Persistence.Repos;
+
+namespace TPP.Core.Commands.Definitions
+{
+    public class PollCommands : ICommandCollection
+    {
+        public IEnumerable<Command> Commands => new[]
+        {
+            new Command("vote", Vote)
+            {
+                Description = "Vote on a poll. Argument: <PollCode> <Option1> <OptionX> (optional if multi-choice poll)"
+            },
+            new Command("poll", Poll)
+            {
+                Aliases = new[] { "checkpoll" },
+                Description = "Check a poll's status and options. Argument: <PollCode>"
+            },
+        };
+
+        private readonly IPollRepo _pollRepo;
+
+        public PollCommands(IPollRepo pollRepo)
+        {
+            _pollRepo = pollRepo;
+        }
+
+        public async Task<CommandResult> Vote(CommandContext context)
+        {
+            //Use poll code and option to vote.
+            //Options can be in ID or String form but must all match
+            //Multi-answer polls use spacing between options
+            //!vote IDENTIFIER Option2 Option4
+            //!vote IDENTIFIER 2 4
+
+            (string pollCode, ManyOf<string> voteStrs) = await context.ParseArgs<string, ManyOf<string>>();
+            ImmutableList<string> votes = voteStrs.Values;
+
+            Poll? poll = await _pollRepo.FindPoll(pollCode);
+            if (poll == null)
+                return new CommandResult { Response = $"No poll with the code '{pollCode}' was found." };
+
+            List<int> selectedOptions = new();
+            foreach (string voteStr in votes)
+            {
+                PollOption? option = null;
+                if (int.TryParse(voteStr.TrimStart('#'), out int voteInt))
```

Comment:
letters (i.e. ABCD) should be valid options imo, though should be regulated in the extremely unlikely/impossible event that there are more than 26 options